plugins/woocommerce/src/Internal/Features/FeaturesController.php

```diff
@@ -545,7 +555,13 @@ function( $feature_id ) use ( $features ) {
 				return $features[ $feature_id ]['is_experimental'];
 			}
 		);
-		$mature_feature_ids       = array_diff( $feature_ids, $experimental_feature_ids );
+		$disabled_ui_feature_ids  = array_filter(
+			$feature_ids,
+			function( $feature_id ) use ( $features ) {
+				return $features[ $feature_id ]['disable_ui'];
+			}
+		);
+		$mature_feature_ids       = array_diff( $feature_ids, $experimental_feature_ids, $disabled_ui_feature_ids );
```

Comment:
We should not exclude features with disabled UI here as `$mature_feature_ids` as that wouldn't prevent experimental features with `disable_ui = true` from having their settings rendered.

My personal suggestion, would be to add a check on [this line](https://github.com/woocommerce/woocommerce/blob/615365d14e0e2b944f05213e366fd1a9849612db/plugins/woocommerce/src/Internal/Features/FeaturesController.php#L595) such as

```php
			if ( isset( $features[ $id ]['disable_ui'] ) && $features[ $id ]['disable_ui'] ) {
				continue;
			}
```

That way, we can skip displaying the setting for features with disabled UI whether it is a mature or experimental feature.

As a side effect, we can get rid of the `$disabled_ui_feature_ids` var, at least for the time being.